dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromPrompt.cs

```diff
@@ -129,21 +128,27 @@ internal sealed class KernelFunctionFromPrompt : KernelFunction
             throw new OperationCanceledException($"A {nameof(Kernel)}.{nameof(Kernel.PromptRendered)} event handler requested cancellation before function invocation.");
         }
 
-        IReadOnlyList<ITextResult> completionResults = await textCompletion.GetCompletionsAsync(renderedPrompt, arguments.ExecutionSettings, kernel, cancellationToken).ConfigureAwait(false);
+        var textContent = await textCompletion.GetTextContentAsync(renderedPrompt, arguments.ExecutionSettings, kernel, cancellationToken).ConfigureAwait(false);
 
-        string completion = await GetCompletionsResultContentAsync(completionResults, cancellationToken).ConfigureAwait(false);
+        var result = new FunctionResult(this, textContent.Text, kernel.Culture, new Dictionary<string, object?>());
+        result.Metadata!.Add(KernelEventArgsExtensions.RenderedPromptMetadataKey, renderedPrompt);
 
-        var modelResults = completionResults.Select(c => c.ModelResult).ToArray();
-
-        return new FunctionResult(
-            this,
-            completion,
-            kernel.Culture,
-            new Dictionary<string, object?>(2)
+        // Merge content metadata with the function result metadata
+        if (textContent.Metadata is not null)
+        {
+            foreach (var kv in textContent.Metadata)
             {
-                [AIFunctionResultExtensions.ModelResultsMetadataKey] = modelResults,
-                [KernelEventArgsExtensions.RenderedPromptMetadataKey] = renderedPrompt,
-            });
+                if (!result.Metadata.ContainsKey(kv.Key))
+                {
+                    result.Metadata.Add(kv.Key, kv.Value);
+                }
+                else
+                {
+                    result.Metadata[kv.Key] = kv.Value;
+                }
+            }
+        }
```

Comment:
The result metadata at this poing has only one key - rendered prompt added above. It seems the whole block for merging can be replaced by this line - `var result = new FunctionResult(this, textContent.Text, kernel.Culture, textContent.Metadata ?? new Dictionary<string, object?>());`

Reply:
Merging is still needed but I simplified the logic using the Dictionary constructor for that.